Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/HDRColorGrading.azsl

```diff
@@ -0,0 +1,212 @@
+    float4 m_smhHighlightsColor;
+
+    // my color grading output
+    float4 m_color;
+}
+
+// Below are the color grading functions. These expect the frame color to be in ACEScg space.
+// If a color parameter is introduced from external sources (e.g. Editor color picker which uses Linear RGB), 
+// it will be converted into ACEScg as needed.
+float3 ColorGradePostExposure (float3 frame_color, float exposure) 
+{
+    frame_color *= pow(2.0f, exposure);
+	return frame_color;
+}
+
+// The contrast equation is performed in ACEScc (logarithmic) color space.
+float3 ColorGradingContrast (float3 frame_color, float midgrey, float amount) 
+{
+    float contrast_adjustment = amount * 0.01f + 1.0f;
+    frame_color = TransformColor(frame_color.rgb, ColorSpaceId::ACEScg, ColorSpaceId::ACEScc);
+    frame_color = (frame_color - midgrey) * contrast_adjustment + midgrey;
+	return frame_color = TransformColor(frame_color.rgb, ColorSpaceId::ACEScc, ColorSpaceId::ACEScg);
+}
+
+// The swatch_color param expects a linear RGB value.
+float3 ColorGradeColorFilter (float3 frame_color, float3 swatch_color, float alpha) 
+{
+    swatch_color = TransformColor(frame_color.rgb, ColorSpaceId::LinearSRGB, ColorSpaceId::ACEScg);
+    swatch_color *= pow(2.0f, PassSrg::m_colorFilterIntensity);
+    float3 frame_adjust = frame_color * swatch_color;
+    return frame_color = lerp(frame_color, frame_adjust, alpha);
+}
+
+float3 ColorGradeHueShift (float3 frame_color, float amount) 
+{
+	float3 frame_hsv = TransformColor(frame_color, ColorSpaceId::ACEScg, ColorSpaceId::HSV);
+	float hue = frame_hsv.x + amount;
+	frame_hsv.x = RotateHue(hue, 0.0, 1.0);
+	return TransformColor(frame_hsv, ColorSpaceId::HSV, ColorSpaceId::ACEScg);
+}
+
+float3 ColorGradeSaturation (float3 frame_color, float control) 
+{
+	float v_luminance = CalculateLuminance(frame_color, ColorSpaceId::ACEScg);
+	return (frame_color - v_luminance) * control + v_luminance;
+}
+
+float3 ColorGradeKelvinColorTemp(float3 frame_color, float kTemp, float mix, float lumPreservation)
+{
+    float3 kColor    = TransformColor(KelvinToRGB(kTemp), ColorSpaceId::LinearSRGB, ColorSpaceId::ACEScg);
+    float3 oLum      = RGBToHSL(frame_color.rgb);
+    float3 blended   = lerp(frame_color.rgb, frame_color.rgb * kColor.rgb, mix);
+    float3 resHSV    = RGBToHSL(blended.rgb);
+    float3 resRGB    = HSLToRGB(float3(resHSV.xy, oLum.z));
+    return lerp(blended.rgb, resRGB.rgb, lumPreservation);
+}
+
+// pow(f, e) won't work if f is negative, or may cause inf/NAN.
+float3 NoNanPow(float3 base, float3 power)
+{
+    return pow(max(abs(base), float3(FLOAT_EPSILON, FLOAT_EPSILON, FLOAT_EPSILON)), power);
+}
+
+float3 ColorGradeSplitTone (float3 frame_color, float balance, float mix) 
+{
+    float3 frame_splitTone = NoNanPow(frame_color, 1.0 / 2.2);
+	float t = saturate(CalculateLuminance(saturate(frame_splitTone), ColorSpaceId::ACEScg) + balance);
+	float3 shadows = lerp(0.5, PassSrg::m_splitToneShadowsColor.rgb, 1.0 - t);
```

Comment:
Remember when these values are exposed in a component later that they should be in ACEScg space (or ACEScc if you switch this whole shader to use ACEScc) by the time they get to the shader. Might be a good idea to comment on that in the pass file that exposes these values. This goes for most if not all of the color values passed in to the shader.

Reply:
Added a comment on the .pass file for color values. The shader is currently expecting linear rgb but perhaps we should convert to ACEScg before sending it into the shader. Do we have functions that do that in C++ already?

Reply:
The shader is mostly expecting ACEScg currently except in cases where you explicitly are converting the value passed in to ACEScg space from some other space. For instance m_splitToneShadowsColor isn't converted so it should be in ACEScg space, but m_colorFilterSwatch is converted and expected to be in sRGB space.

Once you're passing these values in from a component though you should be able to do all the conversions in C++ and not in the shader.

Reply:
Good point, let me add a task to move those calculations into C++, and have the shader expect ACEScg color space inputs. This will require us to add those conversion matrices in C++ as I do not see them anywhere yet.